airbyte-workers/src/main/java/io/airbyte/workers/temporal/scheduling/ConnectionManagerWorkflowImpl.java

```diff
@@ -334,4 +365,34 @@ private void continueAsNew(final ConnectionUpdaterInput connectionUpdaterInput)
     }
   }
 
+  private <INPUT, OUTPUT> OUTPUT runActivityWithOutput(Function<INPUT, OUTPUT> mapper, INPUT input) {
+    try {
+      return mapper.apply(input);
+    } catch (Exception e) {
+      log.error("Failed to run an activity for the connection " + connectionId, e);
+      workflowState.setStuck(true);
+      workflowState.setRetryFailedActivity(false);
+      Workflow.await(() -> workflowState.isRetryFailedActivity());
+      log.error("Retrying an activity for the connection " + connectionId, e);
+      workflowState.setStuck(false);
+      workflowState.setRetryFailedActivity(false);
+      return runActivityWithOutput(mapper, input);
+    }
+  }
+
+  private <INPUT> void runActivity(Consumer<INPUT> consumer, INPUT input) {
+    try {
+      consumer.accept(input);
+    } catch (Exception e) {
+      log.error("Failed to run an activity for the connection " + connectionId, e);
+      workflowState.setStuck(true);
+      workflowState.setRetryFailedActivity(false);
+      Workflow.await(() -> workflowState.isRetryFailedActivity());
+      log.error("Retrying an activity for the connection " + connectionId, e);
+      workflowState.setStuck(false);
+      workflowState.setRetryFailedActivity(false);
+      runActivity(consumer, input);
+    }
+  }
+
```

Comment:
can we dry this up. the duplicate code here is going to be easy to have go out of sync.
```suggestion
  private <INPUT> void runActivity2(final Consumer<INPUT> consumer, final INPUT input) {
    // make the consumer look like a function so that we can reuse logic in runActivityWithOutput
    runActivityWithOutput((inputInternal) -> {
      consumer.accept(inputInternal);
      return null;
    }, input);
  }
```

Reply:
Done